src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -935,6 +935,40 @@ public String getUser() {
         }
     }
 
+    /**
+     * Get a Map of default tab names to deafult tab fields.
+     * The fields are returned as a String with fields separated by ;
+     * They are combined into one string in order to feed into GenFieldsCustomizerDialogViewModel.resetFields()
+     * so that they do not have to be parsed in order to fit there
+     *
+     * @param kesfjaksd;fjlkasdjflkheck.
+     * @return true sfdjklsdjfklse otherwise.
+     */
+    @Override
+    public Map<String, String> getCustomTabsNamesAndFields() {
+        Map<String, String> customTabsMap = new HashMap<>();
+
+        int defNumber = 0;
+        while (true) {
+            //Saved as CUSTOMTABNAME_def{number} and ; separated
+            String name = (String) defaults.get(CUSTOM_TAB_NAME + "_def" + defNumber);
```

Comment:
A better approach is to use two string lists, one for the keys and one for the values.
Have a look at the methods for `setMainTableColumnSortType` and `getMainTableColumnSortTypes` that's where I store the columns and their sort types.

Reply:
Okay. What would the correct data structure be? From my understanding, there are multiple tabs, each of which has a name and multiple fields. How would you mark which fields belonged to which tabs?

Reply:
Hmm that is a thing I didn't have in mind